bitclock-fw/main/libs/mqtt.c

```diff
@@ -0,0 +1,100 @@
+#include "esp_log.h"
+#include "mqtt_client.h"
+#include "mqtt.h"
+
+static const char *TAG = "mqtt";
+
+esp_mqtt_client_handle_t client;
+static bool configs_published = false;
+
+const char *co2_unique_id = "bitclock_co2";
+const char *nox_unique_id = "bitclock_nox";
+const char *voc_unique_id = "bitclock_voc";
+const char *temp_unique_id = "bitclock_temp";
+const char *humidity_unique_id = "bitclock_humidity";
+
+
+bool publish_message(const char *topic, const char *message, bool retain) {
+    bool success = esp_mqtt_client_publish(client, topic, message, 0, 1, retain) >= 0;
+    if (success) {
+      ESP_LOGI(TAG, "MQTT publish successful, topic=%s message=%s", topic, message);
+    } else { 
+      ESP_LOGE(TAG, "MQTT publish failed, topic=%s message=%s", topic, message);
+    }
+    return success;
+}
+
+
+bool publish_homeassistant_config(const char * name, const char * unique_id, const char * device_class, const char * unit_of_measurement, const char * state_class, const char * platform) {
+    char * config_message = malloc(300);
```

Comment:
Can you use a `static char[300]` here or globally above? This `malloc` calls in this file are leaking and probably simpler not to dynamically allocate ever time.